cc_plugin_ncei/ncei_base.py

```diff
@@ -557,50 +541,254 @@ def check_global_attributes(self, dataset):
         :metadata_link = "" ; //............................................. RECOMMENDED - This attribute provides a link to a complete metadata record for this data set or the collection that contains this data set. (ACDD)
         '''
 
-        should_exist = [
+        recommended_ctx = TestCtx(BaseCheck.MEDIUM, 'Recommended global attributes')
+
+        # Do any of the variables define platform ?
+        variable_defined_platform = any((hasattr(var, 'platform') for var in dataset.variables))
+        if not variable_defined_platform:
+            platform_name = getattr(dataset, 'platform', '')
+            recommended_ctx.assert_true(platform_name and platform_name in dataset.variables, 'platform should exist and point to a variable.')
+
+        sea_names = [sn.lower() for sn in util.get_sea_names()]
+        sea_name = getattr(dataset, 'sea_name', '')
+        sea_name = sea_name.replace(', ', ',')
+        sea_name = sea_name.split(',') if sea_name else []
+        for sea in sea_name:
+            recommended_ctx.assert_true(
+                sea.lower() in sea_names,
+                'sea_name attribute should exist and should be from the NODC sea names list: {} is not a valid sea name'.format(sea)
+            )
+
+        # Source: http://www.pelagodesign.com/blog/2009/05/20/iso-8601-date-validation-that-doesnt-suck/
+        iso8601_regex = r'^([\+-]?\d{4}(?!\d{2}\b))((-?)((0[1-9]|1[0-2])(\3([12]\d|0[1-9]|3[01]))?|W([0-4]\d|5[0-2])(-?[1-7])?|(00[1-9]|0[1-9]\d|[12]\d{2}|3([0-5]\d|6[1-6])))([T\s]((([01]\d|2[0-3])((:?)[0-5]\d)?|24\:?00)([\.,]\d+(?!:))?)?(\17[0-5]\d([\.,]\d+)?)?([zZ]|([\+-])([01]\d|2[0-3]):?([0-5]\d)?)?)?)?$'
```

Comment:
I think I'm the one that used the insane regex but using a third party module that the other checkers used would be better if only for the consistency.